specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/networkInterfaces.json

```diff
@@ -0,0 +1,480 @@
+              "$ref": "#/definitions/networkinterfacesListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "InterfaceDNSSettings": {
+      "type": "object",
+      "properties": {
+        "dnsServers": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "List of DNS server IP Addresses for the interface"
+        }
+      }
+    },
+    "networkinterfacesProperties": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "properties": {
+        "ipConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ipConfiguration"
+          },
+          "description": "IPConfigurations - A list of IPConfigurations of the network interface.",
+          "x-ms-identifiers": []
+        },
+        "macAddress": {
+          "type": "string",
+          "description": "MacAddress - The MAC address of the network interface."
+        },
+        "dnsSettings": {
+          "$ref": "#/definitions/InterfaceDNSSettings",
+          "description": "DNS Settings for the interface"
+        },
+        "provisioningState": {
+          "type": "string"
+        },
+        "resourceName": {
+          "type": "string",
+          "description": "name of the object to be used in moc"
+        },
+        "status": {
+          "type": "object",
+          "description": "MOCNetworkInterfaceStatus defines the observed state of MOCNetworkInterface"
+        }
+      },
+      "description": "MOCNetworkInterfaceSpec defines the desired state of MOCNetworkInterface"
+    },
+    "ipConfiguration": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name - The name of the resource that is unique within a resource group. This name can be used to access the resource."
+        },
+        "properties": {
+          "type": "object",
+          "properties": {
+            "gateway": {
+              "x-ms-client-flatten": true,
+              "type": "string",
+              "description": "Gateway for network interface"
+            },
+            "prefixLength": {
+              "type": "string",
+              "description": "prefixLength for network interface"
+            },
+            "privateIPAddress": {
+              "type": "string",
+              "description": "PrivateIPAddress - Private IP address of the IP configuration."
+            },
+            "privateIPAllocationMethod": {
+              "type": "string",
+              "description": "PrivateIPAllocationMethod - The private IP address allocation method. Possible values include: 'Static', 'Dynamic'"
+            },
+            "subnet": {
+              "type": "object",
+              "properties": {
+                "id": {
```

Comment:
>id [](http://example.com/codeflow?start=17&length=2)

Recommend the name resourceId for ID references.

Reply:
Aggreed. The API was published to our customers with the "id" property name (this API was copied from RPSaaSdev, where it was reviewed before). In order to avoid introducing any breaking change, we'd like to keep "id" in this version of the API and fix this in a future revision. Does this plan sound good?

I added this workitem to the pending task here: https://dev.azure.com/msazure/msk8s/_boards/board/t/ARC-HCI/Stories/?workitem=12679581